code/game/objects/items/devices/scanners.dm

```diff
@@ -966,3 +967,65 @@ GENE SCANNER
 	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 	return TRUE
 
+
+/obj/item/material_scanner
+	name = "material scanner"
+	desc = "A device that analyzes a object's composition and lists its materials."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "forensic0"
+	item_state = "analyzer"
+	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	flags_1 = CONDUCT_1
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	materials = list(/datum/material/iron=30, /datum/material/glass=20)
+
+/obj/item/material_scanner/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(istype(target,/obj/item))
+		var/obj/item/I = target
+		if (I.materials && length(I.materials)>0)
+			analyze_materials(I, user)
+			return
+	to_chat(user, "<span class='notice'>[target] doesn't have any reusable materials.</span>")
+
+/obj/item/material_scanner/proc/analyze_materials(obj/item/I, mob/living/user)
+	to_chat(user, "- Material Analysis of [I] -")
+	var/integ = 1
+	var/output = ""
+	for(var/R in I.materials)
+		var/datum/material/M = R
+		var/amount =  I.materials[M]
+		var/fcolor = "#696969"	//hehehe
+		if (istype(M,/datum/material/glass))
+			fcolor = "#60b6ff"
+		else if (istype(M,/datum/material/gold))
+			fcolor = "#ffcc4f"
+		else if (istype(M,/datum/material/silver))
+			fcolor = "#ccccdb"
+		else if (istype(M,/datum/material/copper))
+			fcolor = "#d57e00"
+		else if (istype(M,/datum/material/diamond))
+			fcolor = "#d5f5ff"
+		else if (istype(M,/datum/material/uranium))
+			fcolor = "#83bf33"
+		else if (istype(M,/datum/material/plasma))
+			fcolor = "#ee82ee"
+		else if (istype(M,/datum/material/bluespace))
+			fcolor = "#0000ff"
+		else if (istype(M,/datum/material/bananium))
+			fcolor = "#ffff00"
+		else if (istype(M,/datum/material/titanium))
+			fcolor = "#dddddd"
+		output+= " - <font color='[fcolor]'>[M.name] ([amount])</font>"
+		integ++
+		if (integ>3)
+			to_chat(user, output + " - ")
+			integ = 1
+			output = ""
```

Comment:
What is this and why?

Reply:
So it outputs:
-mat 1 - mat 2 - mat 3 -
-mat 4 - mat 5 - mat 6 -

instead of
-mat 1 - mat 2 - mat 3 - mat 4 - mat 5 - mat 6 -